x/cdp/keeper/cdp.go

```diff
@@ -340,72 +341,73 @@ func (k Keeper) SetGovDenom(ctx sdk.Context, denom string) {
+func (k Keeper) ValidateCollateral(ctx sdk.Context, collateral sdk.Coins) error {
 	if len(collateral) != 1 {
-		return types.ErrInvalidCollateralLength(k.codespace, len(collateral))
+		return sdkerrors.Wrapf(types.ErrInvalidCollateralLength, "%d", len(collateral))
 	}
 	_, found := k.GetCollateral(ctx, collateral[0].Denom)
 	if !found {
-		return types.ErrCollateralNotSupported(k.codespace, collateral[0].Denom)
+		return sdkerrors.Wrap(types.ErrCollateralNotSupported, collateral[0].Denom)
 	}
 	return nil
 }
 
 // ValidatePrincipalAdd validates that an asset is valid for use as debt when creating a new cdp
-func (k Keeper) ValidatePrincipalAdd(ctx sdk.Context, principal sdk.Coins) sdk.Error {
+func (k Keeper) ValidatePrincipalAdd(ctx sdk.Context, principal sdk.Coins) error {
 	for _, dc := range principal {
 		dp, found := k.GetDebtParam(ctx, dc.Denom)
 		if !found {
-			return types.ErrDebtNotSupported(k.codespace, dc.Denom)
+			return sdkerrors.Wrap(types.ErrDebtNotSupported, dc.Denom)
 		}
 		if dc.Amount.LT(dp.DebtFloor) {
-			return types.ErrBelowDebtFloor(k.codespace, sdk.NewCoins(dc), dp.DebtFloor)
+			return sdkerrors.Wrapf(types.ErrBelowDebtFloor, "proposed %s < minimum %s", sdk.NewCoins(dc), dp.DebtFloor)
 		}
 	}
 	return nil
 }
 
 // ValidatePrincipalDraw validates that an asset is valid for use as debt when drawing debt off an existing cdp
-func (k Keeper) ValidatePrincipalDraw(ctx sdk.Context, principal sdk.Coins) sdk.Error {
+func (k Keeper) ValidatePrincipalDraw(ctx sdk.Context, principal sdk.Coins) error {
 	for _, dc := range principal {
 		_, found := k.GetDebtParam(ctx, dc.Denom)
 		if !found {
-			return types.ErrDebtNotSupported(k.codespace, dc.Denom)
+			return sdkerrors.Wrap(types.ErrDebtNotSupported, dc.Denom)
 		}
 	}
 	return nil
 }
 
 // ValidateDebtLimit validates that the input debt amount does not exceed the global debt limit or the debt limit for that collateral
-func (k Keeper) ValidateDebtLimit(ctx sdk.Context, collateralDenom string, principal sdk.Coins) sdk.Error {
+func (k Keeper) ValidateDebtLimit(ctx sdk.Context, collateralDenom string, principal sdk.Coins) error {
 	cp, found := k.GetCollateral(ctx, collateralDenom)
 	if !found {
-		return types.ErrCollateralNotSupported(k.codespace, collateralDenom)
+		return sdkerrors.Wrap(types.ErrCollateralNotSupported, collateralDenom)
 	}
+
 	for _, dc := range principal {
 		totalPrincipal := k.GetTotalPrincipal(ctx, collateralDenom, dc.Denom).Add(dc.Amount)
 		collateralLimit := cp.DebtLimit.AmountOf(dc.Denom)
 		if totalPrincipal.GT(collateralLimit) {
-			return types.ErrExceedsDebtLimit(k.codespace, sdk.NewCoins(sdk.NewCoin(dc.Denom, totalPrincipal)), sdk.NewCoins(sdk.NewCoin(dc.Denom, collateralLimit)))
+			return sdkerrors.Wrapf(types.ErrExceedsDebtLimit, "payment %s > debt %s", sdk.NewCoins(sdk.NewCoin(dc.Denom, totalPrincipal)), sdk.NewCoins(sdk.NewCoin(dc.Denom, collateralLimit)))
```

Comment:
nit: Error message is a bit deceiving. The individual payment isn't greater than the outstanding debt limit, the payment puts the outstanding debt above the limit.